### DISCUSSION TITLE
counting the integer operations on a Genoa cpu

### DISCUSSION THREAD

**Discussion by meta1360** [amd/amd-lab-notes#19] [General]
Hi, I am trying to capture the number of integer operations as a certain application is running on a Genoa CPU. if you know the command in uprof or event ID/mask that I can run with perf stat it would be great.

**Comment by naipatil** [comment]
Hello, 

Can you please try -
./AMDuProfCLI collect -e event=pmcx0ab,umask=0x8 -e event=pmcx008,umask=0xf -e event=pmcx00d,umask=0xF0 -o <output-folder> <application>

**Comment by meta1360** [comment]
Hi @naipatil 
Thanks for your reply. 
Which one is for integer scalar?
Is it correct to say that the first event is related to integer vector and the last one is for floating-point? right?
The second gives me all scalar integer operations? 
I am running it on a stream app with 125M integer entries to do a triad kernel `(c[I] = a[I] + cte * b[I])` for 200 iterations, I try to ensure that the GCC compiler uses integer scalar for this array (using -O, instead of -O3). 
I am expecting the number of integer operations is 125M * 200 * 2 but the second event captures as high as ~1.8x to ~2.0x of my expectation. 

- packed_int_op_type.int256_all  --> Umask = 0xF0 event = 0x0d

- de_dis_ops_from_decoder.disp_op_type.any_integer_dispatch  --> Umask = 0x08 event = 0xab

- fp_ops_retired_by_type.scalar_all  --> Umask = 0x0F event = 0x0a
